### PR TITLE
etcd client atomic storage api

### PR DESCRIPTION
**Pull request by stellarspot**
* Start etcd storage by snet-daemon according to the config
* Implement atomic storage api by etcd client

### CODE REVIEW

File: escrow/etcd_storage_test.go [hunk 1]
```diff
@@ -165,31 +154,6 @@ func getFreePort() (port int, err error) {
...
-	return &handler.GrpcStreamContext{
-		MD: md,
-	}
-}
-
-func clearTestEtcdContext() {
-	cleanableEtcdStorage.Clear()
-}
-
 func TestEtcdGetPayment(t *testing.T) {
```
Comment (vsbogd) [inline escrow/etcd_storage_test.go]: TestEtcdGetPayment -> TestEtcdGet?

File: etcddb/etcddb_client.go [hunk 2]
```diff
@@ -123,15 +125,45 @@ func (client *EtcdClient) CompareAndSet(key []byte, expect []byte, update []byte
...
+	defer cancel()
+
+	etcdv3 := client.etcdv3
+	session, err := concurrency.NewSession(etcdv3)
+
+	if err != nil {
+		return
+	}
+
+	mu := concurrency.NewMutex(session, key)
```
Comment (vsbogd) [inline etcddb/etcddb_client.go]:
```suggestion
	// TODO: implement it using etcdv3.KV.Txn(ctx).If(...).Then(...).Commit() as in CompareAndSwap()
	mu := concurrency.NewMutex(session, key)
```

File: escrow/etcd_storage_test.go (hunk 1 above)
Comment (stellarspot) [reply escrow/etcd_storage_test.go]: I deleted the etcd_storage_test.go test.

File: snetd/cmd/serve.go [hunk 3]
```diff
@@ -139,10 +141,24 @@ func newDaemon() (daemon, error) {
 		d.sslCert = &cert
 	}
 
+	etcdServer, err := etcddb.GetEtcdServer()
```
[the remark this answers is not in the document]
Comment (stellarspot) [reply snetd/cmd/serve.go]: Updated in #80

**Pull request merged**